### PR TITLE
Add ImageCatalog post-install hook

### PR DESCRIPTION
**Pull request by stone-z**
Towards https://github.com/giantswarm/giantswarm/issues/30093

This PR:

- deploys a ClusterImageCatalog to allow consumers to use GS-retagged images
- sets a VPA minimum CPU

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: helm/cloudnative-pg/templates/image-catalog.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: gs-postgresql
+  annotations:
+    "helm.sh/hook": "post-install,post-upgrade"
+  labels:
+    {{- include "cloudnative-pg.labels" . | nindent 4 }}
+spec:
+  images:
+    - major: 16
+      image: quay.io/giantswarm/postgresql-cnpg:16.2
```
Comment (fhielpos) [inline helm/cloudnative-pg/templates/image-catalog.yaml:12]: Why quay? and can we template the registry?

Comment (stone-z) [reply helm/cloudnative-pg/templates/image-catalog.yaml:12]: The image isn't being crsynced for some reason. The registry will definitely be templated once that's resolved

Comment (stone-z) [reply helm/cloudnative-pg/templates/image-catalog.yaml:12]: Actually, it seems it is synced to aliyun and azure, just not Docker :thinking:

Comment (stone-z) [reply helm/cloudnative-pg/templates/image-catalog.yaml]: This turned out to be a quirk of crsync. I've updated the catalog to template in the registry

**Pull request merged**